DSharpPlus/Entities/Channel/Message/DiscordMessageBuilder.cs

```diff
@@ -285,6 +304,12 @@ internal void Validate(bool isModify = false)
             {
                 if (this.Files?.Count == 0 && string.IsNullOrEmpty(this.Content) && this.Embed == null)
                     throw new ArgumentException("You must specify content, an embed, or at least one file.");
+
+                if (this.Components.Count + 1 > 5)
+                    throw new InvalidOperationException("You can only have 5 action rows per message.");
```

Comment:
I'm not too sure either; I should be doing `if (this.Components.Count is 5)`

Reply:
Are you being sarcastic? That would also be a bug, this is the Validate method not the Add method, meaning it could be dealing with a list that has more than 5 elements.

Reply:
You still need to change this, likely to `this.Components.Count > 5`.